src/Nethermind/Nethermind.Evm/CodeAnalysis/StatsAnalyzer.cs

```diff
@@ -0,0 +1,201 @@
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace Nethermind.Evm.CodeAnalysis
+{
+    public class StatsAnalyzer
+    {
+        private static StatsAnalyzer? _instance;
+        private static readonly object _lock = new object();
+
+        private SortedSet<(ulong, uint)> _topN;
```

Comment:
My understanding of using the `TopN` is that you don't need the `SortedSet` but rather a heap. If this is the case, `PriorityQueue` from BCL that uses anary trees underneath should be faster and more aligned?